analyses/cell-type-neuroblastoma-04/run-analysis.sh

```diff
@@ -51,32 +75,58 @@ download_file() {
+nbatlas_full_seurat="${scratch_dir}/seuratObj_NBAtlas_share_v20241203.rds" # this is needed regardless of which nbatlas_version was specified since we need to subset ids
+nbatlas_tumor_metadata_file="${scratch_dir}/SeuratMeta_Share_TumorZoom_NBAtlas_v20250228.rds"
 
-download_file $nbatlas_subset_url $nbatlas_subset_seurat
+# Define which NBAtlas version we're actually converting, and download subset if it's that one
+if [[ $nbatlas_version == "full" ]]; then
+  nbatlas_seurat=${nbatlas_full_seurat}
+else
+  nbatlas_seurat="${scratch_dir}/SeuratObj_Share_50kSubset_NBAtlas_v20240130.rds"
+  download_file $nbatlas_subset_url $nbatlas_seurat
+fi
+
+# grab remaining downloads
 download_file $nbatlas_full_url $nbatlas_full_seurat
 download_file $nbatlas_tumor_url $nbatlas_tumor_metadata_file
 
-# Convert the NBAtlas object to SCE and AnnData formats, for both the full and subsetted versions
-cell_id_file="${scratch_dir}/nbtalast-cell-ids.txt.gz"
+
+# Determine the ids to retain based on the *full* atlas object
+cell_id_file="${scratch_dir}/nbatlas-cell-ids.txt.gz"
 Rscript ${script_dir}/00a_extract-nbatlas-ids.R \
     --nbatlas_file "${nbatlas_full_seurat}" \
     --cell_id_file "${cell_id_file}"
 
+# Convert the NBAtlas object to SCE (and in the future, also to AnnData)
 Rscript ${script_dir}/00b_convert-nbatlas.R \
-    --nbatlas_file "${nbatlas_subset_seurat}" \
-    --tumor_metadata_file "${nbatlas_tumor_metadata_file}" \
-    --cell_id_file "${cell_id_file}" \
-    --sce_file "${nbatlas_subset_sce}" \
-    --anndata_file "${nbatlas_subset_anndata}" \
-    ${test_flag}
+   --nbatlas_file "${nbatlas_seurat}" \
+   --tumor_metadata_file "${nbatlas_tumor_metadata_file}" \
+   --cell_id_file "${cell_id_file}" \
+   --sce_file "${nbatlas_sce}" \
+   ${test_flag}
+   # For now, we will not save the AnnData object
+   #--anndata_file "${nbatlas_anndata}"
 
-Rscript ${script_dir}/00b_convert-nbatlas.R \
-    --nbatlas_file "${nbatlas_full_seurat}" \
-    --tumor_metadata_file "${nbatlas_tumor_metadata_file}" \
-    --cell_id_file "${cell_id_file}" \
-    --sce_file "${nbatlas_full_sce}" \
-    --anndata_file "${nbatlas_full_anndata}" \
-    ${test_flag}
+
+
+###################################################################
+######################## SingleR annotation #######################
+###################################################################
+
+echo "Training SingleR model"
+
+# Train the SingleR model with an aggregated reference
+singler_model_file="${scratch_dir}/singler-model_aggregated_nbatlas-${nbatlas_version}.rds"
+
+# can pass in an arbitrary SCE here for sce_file; this is just the first sample in the project
+Rscript ${script_dir}/01_train-singler-model.R \
+    --nbatlas_sce "${nbatlas_sce}" \
+    --sce_file "${data_dir}/SCPCS000101/SCPCL000118_processed.rds" \
+    --singler_model_file "${singler_model_file}" \
+    --aggregate_reference
```

Comment:
The same logic should be applied here where we always do the subset and if testing is false we do the full. I would maybe group the if statement and first process the subset including conversion and creating the model and then check the testing flag and if it's false convert and train the full model.

Reply:
@allyhawkins I do actually want to confirm you seen this reply here, since it means we can _get rid of_ the `00a_extract-nbatlas-ids.R` script and behavior. I think I should formally remove that script in a separate PR though, but in this PR I'll get it _only downloading EITHER_ the full or subset atlas. Does this sound ok?

Reply:
> @allyhawkins I do actually want to confirm you seen this reply here, since it means we can _get rid of_ the `00a_extract-nbatlas-ids.R` script and behavior. I think I should formally remove that script in a separate PR though, but in this PR I'll get it _only downloading EITHER_ the full or subset atlas. Does this sound ok?

Yes I think your statement about only using the full atlas for actual analysis moving forward makes sense. So I agree that we should update this to download either the full (default) or the subset (for testing) only. I think you could still just use the `testing` flag that you have instead of specifying a version though.